pkg/api/owners.go

```diff
@@ -0,0 +1,687 @@
+	return o.OwnersFor(id).SimpleOwner(Field_CPURealtimeRuntime.Key())
+}
+
+func (o *OwningPlugins) CPURealtimePeriodOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(Field_CPURealtimePeriod.Key())
+}
+
+func (o *OwningPlugins) CPUSetCPUsOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(Field_CPUSetCPUs.Key())
+}
+
+func (o *OwningPlugins) CPUSetMemsOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(Field_CPUSetMems.Key())
+}
+
+func (o *OwningPlugins) PidsLimitOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(Field_PidsLimit.Key())
+}
+
+func (o *OwningPlugins) HugepageLimitOwner(id, size string) (string, bool) {
+	return o.OwnersFor(id).CompoundOwner(Field_HugepageLimits.Key(), size)
+}
+
+func (o *OwningPlugins) BlockioClassOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(Field_BlockioClass.Key())
+}
+
+func (o *OwningPlugins) RdtClassOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(Field_RdtClass.Key())
+}
+
+func (o *OwningPlugins) CgroupsUnifiedOwner(id, key string) (string, bool) {
+	return o.OwnersFor(id).CompoundOwner(Field_CgroupsUnified.Key(), key)
+}
+
+func (o *OwningPlugins) CgroupsPathOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(Field_CgroupsPath.Key())
+}
+
+func (o *OwningPlugins) OomScoreAdjOwner(id string) (string, bool) {
+	return o.OwnersFor(id).SimpleOwner(Field_OomScoreAdj.Key())
+}
+
+func (o *OwningPlugins) RlimitOwner(id, typ string) (string, bool) {
+	return o.OwnersFor(id).CompoundOwner(Field_Rlimits.Key(), typ)
+}
+
+func (o *OwningPlugins) OwnersFor(id string) *FieldOwners {
+	f, ok := o.Owners[id]
+	if !ok {
+		f = NewFieldOwners()
+		o.Owners[id] = f
+	}
+	return f
+}
+
+func NewFieldOwners() *FieldOwners {
+	return &FieldOwners{
+		Simple:   make(map[int32]string),
+		Compound: make(map[int32]*CompoundFieldOwners),
+	}
+}
+
+func (f *FieldOwners) IsCompoundConflict(field int32, key, plugin string) error {
+	m, ok := f.Compound[field]
+	if !ok {
+		f.Compound[field] = NewCompoundFieldOwners()
+		return nil
+	}
+
+	other, claimed := m.Owners[key]
+	if !claimed {
+		return nil
+	}
+
+	clearer, ok := IsMarkedForRemoval(other)
+	if ok {
+		if clearer == plugin {
+			return nil
+		}
+		other = clearer
+	}
+
+	return f.Conflict(field, plugin, other, key)
+}
+
+func (f *FieldOwners) IsSimpleConflict(field int32, plugin string) error {
+	other, claimed := f.Simple[field]
+	if !claimed {
+		return nil
+	}
+
+	clearer, ok := IsMarkedForRemoval(other)
+	if ok {
+		if clearer == plugin {
+			return nil
+		}
+		other = clearer
+	}
+
+	return f.Conflict(field, plugin, other)
+}
+
+func (f *FieldOwners) ClaimCompound(field int32, key, plugin string) error {
+	if err := f.IsCompoundConflict(field, key, plugin); err != nil {
+		return err
+	}
+
+	f.Compound[field].Owners[key] = plugin
+	return nil
+}
+
+func (f *FieldOwners) ClaimSimple(field int32, plugin string) error {
```

Comment:
This doesn't appear to be called outside `owners.go` or even outside of `FieldOwners` methods.  Can it be made private?

Reply:
Made it private.